File: 2019카카오개발자겨울인턴십/크레인인형뽑기게임.js
function solution(board, moves) {
    var answer = 0;
    const arr = []
    moves.forEach(a => {
        const doll = pickup(board, a - 1);
        if(doll){
            if(arr[arr.length - 1] === doll){
                arr.pop();
                answer += 2;
            } else {
                arr.push(doll);
            }
        }
        
    });
    console.log(answer)
    return answer;
}

function pickup(board, index) {
    const nboard = board;
    for (let i = 0; i < nboard.length; i++) {
        if (nboard[i][index] !== 0) {
            const doll = nboard[i][index];
            nboard[i][index] = 0;
            return doll;    
        }
    }
}

solution([[0,0,0,0,0],[0,0,1,0,3],[0,2,5,0,1],[4,2,4,4,2],[3,5,1,3,1]], [1,5,3,5,1,2,1,4])

File: 2019카카오개발자겨울인턴십/크레인인형뽑기게임.py
def solution(board, moves):
    answer = 0
    stack = []
    N = len(board)
    for mo in moves:
        for i in range(N):
            if board[i][mo - 1] != 0:
                if stack == []:
                    stack.append(board[i][mo - 1])
                elif stack[-1] == board[i][mo - 1]:
                    answer += 2
                    stack.pop()
                else:
                    stack.append(board[i][mo - 1])
                board[i][mo - 1] = 0
                break
                
    return answer


# 바꾼게 하나밖에 없는데 시간차이는 꽤 난다.
def solution(board, moves):
    answer = 0
    stack = []
    N = len(board)
    for mo in moves:
        mo -= 1
        for i in range(N):
            if board[i][mo] != 0:
                if stack == []:
                    stack.append(board[i][mo])
                elif stack[-1] == board[i][mo]:
                    answer += 2
                    stack.pop()
                else:
                    stack.append(board[i][mo])
                board[i][mo] = 0
                break
                
    return answer